### ISSUE TITLE
Validator does not allow negative values for floats

### ISSUE THREAD

**Issue by philrittenhouse** [homieiot/convention-website#51]
I get the following error 

`Property 'pressure' value of type float contains invalid data: '-73'!`

when validating the following

```
homie/SPG9f374743ac/$name Smart Pool Gauge
homie/SPG9f374743ac/$homie 4.0.0
homie/SPG9f374743ac/$state ready
homie/SPG9f374743ac/$nodes gauge
homie/SPG9f374743ac/$extensions ""
homie/SPG9f374743ac/gauge/$name Gauge node
homie/SPG9f374743ac/gauge/$type ""
homie/SPG9f374743ac/gauge/$properties pressure,speed,alert,battery
homie/SPG9f374743ac/gauge/pressure -73
homie/SPG9f374743ac/gauge/pressure/$name pressure
homie/SPG9f374743ac/gauge/pressure/$datatype float
homie/SPG9f374743ac/gauge/pressure/$unit psi
```

Is there any reason negative values would not be allowed?  I would assume so but the docs do actually say

> Floats range from 2-1074 to (2-2-52)*21023

which is a somewhat garbled version of what's in the .md file: 2E-1074 to (2 - 2E-52) * 2E1023. 
In both cases I think they should be prefixed by a +/-

I think the fix is simple enough.  It looks like the optional minus sign is missing from the regex in the code:

```
          case "float":
                            if (!value.match(/^[\d\.]+$/)) 
```

Thanks,
Phil

**Comment by davidgraeff** [comment]
Hey Phil,

could you make an MR that fixes this? If you want to, please also create an MR for the specification to officially allow negative floats. I assume you will get approval very fast, it doesn't look like this restriction was intentional.

**Comment by philrittenhouse** [comment]
Sure.  Let me take a crack at it.

**Comment by philrittenhouse** [comment]
I don't have a access to a 64-bit linux box needed to run the extended hugo, so I'm not able to rebuild and test the changes to the website.  Do you still want me to generate a pull request?

**Comment by davidgraeff** [comment]
The CI should hint you at errors if there are any. No access to Linux, that sounds horrible xD

**Comment by philrittenhouse** [comment]
PR #52 for float validation is ready

**Comment by philrittenhouse** [comment]
PR for the convention.md change is PR #217

**Comment by Thalhammer** [comment]
Closed by #52
